### PR TITLE
Publish user registration event

### PR DESCRIPTION
**Pull request by nk-coding**
https://frontend.gropius.duckdns.org/components/9de999cf-58cd-4b47-aa0e-860cd2b6869c/issues/b87c74cb-d40e-4fca-8122-a4593d236015

DoD
- [x] Requirements of the issue are met
- [x] Code has been reviewed
- [x] Code is documented
- [x] GraphQL related artefacts are documented

### CODE REVIEW

File: spi-extension/pom.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.misarch.keycloak.event.provider</groupId>
+    <artifactId>keycloak-dapr-events</artifactId>
+    <version>1.0.0</version>
+    <packaging>jar</packaging>
+
+    <name>Keycloak: Event Publisher to dapr</name>
+    <properties>
+        <keycloak.version>23.0.6</keycloak.version>
```
Comment (delvh) [inline spi-extension/pom.xml:12]: I can foresee problems when updating keycloak as devs might not know they need to update both `Dockerfile` and this `pom.xml`.
Perhaps we should document that in the `README`?

File: spi-extension/src/main/java/org/misarch/keycloak/event/provider/DaprEventListenerProvider.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    /**
+     * Handles user registration.
+     *
+     * @param realmId id of the realm where the user was registered
+     * @param userId id of the registered user
+     */
+    private void handleUserRegister(final String realmId, final String userId) {
+        log.info("User registered: " + realmId + ", " + userId);
```
Comment (delvh) [inline spi-extension/src/main/java/org/misarch/keycloak/event/provider/DaprEventListenerProvider.java]:
```suggestion
        log.infof("User %s in realm %s was registered", userId, realmId);
```

File: spi-extension/src/main/java/org/misarch/keycloak/event/provider/DaprEventListenerProvider.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        KeycloakModelUtils.runJobInTransaction(session.getKeycloakSessionFactory(), session -> {
+            final RealmModel realm = session.realms().getRealm(realmId);
+            final UserModel user = session.users().getUserById(realm, userId);
+            log.info("User: " + user.getUsername() + ", " + user.getEmail() + ", " + user.getFirstName() + ", "
+                    + user.getLastName());
```
Comment (delvh) [inline spi-extension/src/main/java/org/misarch/keycloak/event/provider/DaprEventListenerProvider.java]:
```suggestion
            log.infof("Registered user [name=\"%s\", email=\"%s\", first name=\"%s\", last name=\"%s\"]", user.getUsername(), user.getEmail(), user.getFirstName(), user.getLastName());
```

File: spi-extension/src/main/java/org/misarch/keycloak/event/provider/DaprEventListenerProviderFactory.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    public void postInit(KeycloakSessionFactory factory) {}
+
+    @Override
+    public void close() {}
+
+    @Override
+    public String getId() {
+        return "keycloak-dapr-events";
+    }
+}
```
Comment (delvh) [inline spi-extension/src/main/java/org/misarch/keycloak/event/provider/DaprEventListenerProviderFactory.java:40]: How I love enterprise Java.

Comment (nk-coding) [reply spi-extension/src/main/java/org/misarch/keycloak/event/provider/DaprEventListenerProviderFactory.java:40]: You at least did not have to write this sh**

File: spi-extension/pom.xml (hunk 1 above)
Comment (nk-coding) [reply spi-extension/pom.xml:12]: added with a3fa0964a328114419bdb9c13e5f5bed0f4a6b81

**Review by delvh — APPROVED** [review]

**Pull request merged**